packages/langium/src/parser/langium-parser.ts

```diff
@@ -466,21 +466,19 @@ export class LangiumParser extends AbstractLangiumParser {
     }
 
     private constructInfix(obj: InfixElement, precedence: Map<string, OperatorPrecedence>): any {
-        if (obj.parts.length === 1) {
+        const parts = obj.parts ?? [];
+        const operators = obj.operators ?? [];
+        if (parts.length < 2) {
```

Comment:
Wouldn't a check like this be sufficient here?
```suggestion
        if (Array.isArray(obj.parts) || obj.parts.length <= 1) {
```
For the remainder of this method (parts >= 2) the assignment to a local variable doesn't make any difference, does it?

Reply:
Update: It needs to be `!Array.isArray(obj.parts)` of course!